spec/labels_checker_spec.rb

```diff
@@ -0,0 +1,173 @@
+# frozen_string_literal: true
+
+require_relative 'spec_helper'
+
+module Danger
+  describe Danger::LabelsChecker do
+    it 'is a plugin' do
+      expect(described_class.new(nil)).to be_a Danger::Plugin
+    end
+
+    describe 'with Dangerfile' do
+      before do
+        @dangerfile = testing_dangerfile
+        @plugin = @dangerfile.labels_checker
+      end
+
+      context 'with required labels' do
+        it 'returns a custom error when a PR does not have at least one label' do
+          allow(@plugin.github).to receive(:pr_labels).and_return([])
+
+          error = 'PR is missing at least one label.'
+          @plugin.check(required_labels: [/.*/], required_labels_error: error)
+
+          expect(@dangerfile.status_report[:warnings]).to be_empty
+          expect(@dangerfile.status_report[:errors]).to eq([error])
+        end
+
+        it 'does nothing when a PR has at least one label' do
+          pr_labels = ['my-label']
+          allow(@plugin.github).to receive(:pr_labels).and_return(pr_labels)
+
+          @plugin.check(required_labels: [/.*/])
+
+          expect(@dangerfile.status_report[:errors]).to be_empty
+          expect(@dangerfile.status_report[:warnings]).to be_empty
+        end
+
+        it 'returns an error containing the required labels when a PR does not meet the label requirements' do
+          pr_label = 'random other label'
+          allow(@plugin.github).to receive(:pr_labels).and_return([pr_label])
+
+          @plugin.check(
+            required_labels: [/feature: .*/]
+          )
+
+          expect(@dangerfile.status_report[:warnings]).to be_empty
+          expect(@dangerfile.status_report[:errors]).to eq(['PR is missing label(s) matching: `feature: .*`'])
+        end
+
+        it 'returns a custom error when a PR has custom label requirements' do
+          pr_label = 'random other label'
+          allow(@plugin.github).to receive(:pr_labels).and_return([pr_label])
+
+          custom_labels_error = "Please add at least one label in the 'feature: myFeatureName' format."
+
+          @plugin.check(
+            required_labels: [/feature: .*/],
+            required_labels_error: custom_labels_error
+          )
+
+          expect(@dangerfile.status_report[:warnings]).to be_empty
+          expect(@dangerfile.status_report[:errors]).to eq([custom_labels_error])
+        end
+
+        it 'does nothing when custom required labels are correctly set in the PR' do
+          pr_labels = [
+            'feature: time travel',
+            'type: prototype'
+          ]
```

Comment:
Just so that this test also ensures that our implementation does not accidentally require **_all_** labels to meet the requirements, but only that we have at least one label meeting each requirement (in other words, that the behavior is not "all labels need to meet the requirements", but "each requirement needs to be met by at least one label")

```suggestion
          pr_labels = [
            'feature: time travel',
            'type: prototype',
            'some other label'
          ]
```